docs/csharp/language-reference/attributes/pseudo-attributes.md

```diff
@@ -0,0 +1,55 @@
+| <xref:System.NonSerializedAttribute?displayProperty=fullName>                        | `notserialized` |           |
+| <xref:System.Runtime.InteropServices.OptionalAttribute?displayProperty=fullName>     | `[opt]`         |           |
+| <xref:System.Runtime.InteropServices.OutAttribute?displayProperty=fullName>          | `[out]`         |           |
+| <xref:System.Runtime.InteropServices.PreserveSigAttribute?displayProperty=fullName>  | `preservesig`   |           |
+| <xref:System.SerializableAttribute?displayProperty=fullName>                         | `serializable`  |           |
+| <xref:System.Runtime.InteropServices.StructLayoutAttribute?displayProperty=fullName> | `auto`, `sequential`, or `explicit` | Layout options can be set using the parameters. |
+| <xref:System.Runtime.CompilerServices.IndexerNameAttribute?displayProperty=fullName> |                 | You add this attribute to an indexer to set a different method name. By default, indexers are compiled to a property named `Item`. You can specify a different name using this attribute. |
+
+Some of these custom attributes are applied using other C# syntax rather than adding the attribute to your source code.
+
+| Attribute                                                                                        | Comments |
+|--------------------------------------------------------------------------------------------------|----------|
+| <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute?displayProperty=fullName>    | Specifies the default value for the parameter. Use the [default parameter syntax](../../methods.md#optional-parameters-and-arguments). |
+| <xref:System.Runtime.InteropServices.InAttribute?displayProperty=fullName>                       | Specifies the IL `[in]` modifier. Use the [`in`](../keywords/method-parameters.md#in-parameter-modifier) or [`ref readonly`](../keywords/method-parameters.md#ref-readonly-modifier). |
+| <xref:System.Runtime.CompilerServices.SpecialNameAttribute?displayProperty=fullName>             | Specifies the IL `specialname` modifier. The compiler automatically adds this modifier for methods that require it. |
+| <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute?displayProperty=nameWithType> | This attribute is required for the `delegate*` feature. The compiler adds it to any [`delegate*`](../unsafe-code.md#function-pointers) that requires its use. |
+
+The following custom attributes are generally disallowed in C# source. They're listed here to aid library authors who use reflection, and to ensure you don't create custom attributes with the same name.
```

Comment:
Though, creating custom attributes with the same name in a different namespace is okay, and custom attributes don't go in the System namespace.